lib/kube/proxy/kube_creds_test.go

```diff
@@ -0,0 +1,274 @@
+		client              dynamicCredsClient
+		validateBearerToken func(string) error
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantAddr string
+	}{
+		{
+			name: "aws eks cluster",
+			args: args{
+				cluster: awsKube,
+				client:  getAWSClientRestConfig(cloudclients),
+				validateBearerToken: func(token string) error {
+					if token == "" {
+						return trace.BadParameter("missing bearer token")
+					}
+					tokens := strings.Split(token, ".")
+					if len(tokens) != 2 {
+						return trace.BadParameter("invalid bearer token")
+					}
+					if tokens[0] != "k8s-aws-v1" {
+						return trace.BadParameter("token must start with k8s-aws-v1")
+					}
+					dec, err := base64.RawStdEncoding.DecodeString(tokens[1])
+					if err != nil {
+						return trace.Wrap(err)
+					}
+					if string(dec) != stsMock.u.String() {
+						return trace.BadParameter("invalid token payload")
+					}
+					return nil
+				},
+			},
+			wantAddr: "api.eks.us-west-2.amazonaws.com:443",
+		},
+		{
+			name: "gcp gke cluster",
+			args: args{
+				cluster:             gkeKube,
+				client:              gcpRestConfigClient(cloudclients),
+				validateBearerToken: func(_ string) error { return nil },
+			},
+			wantAddr: "api.gke.google.com:443",
+		},
+		{
+			name: "azure aks cluster",
+			args: args{
+				cluster:             aksKube,
+				client:              azureRestConfigClient(cloudclients),
+				validateBearerToken: func(_ string) error { return nil },
+			},
+			wantAddr: "api.aks.microsoft.com:443",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClock := clockwork.NewFakeClock()
+			got, err := newDynamicKubeCreds(
+				context.Background(),
+				dynamicCredsConfig{
+					clock: fakeClock,
+					checker: func(_ context.Context, _ string,
+						_ authztypes.SelfSubjectAccessReviewInterface,
+					) error {
+						return nil
+					},
+					log:         log,
+					kubeCluster: tt.args.cluster,
+					client:      tt.args.client,
+				},
+			)
+			require.NoError(t, err)
+			require.Equal(t, got.getKubeRestConfig().CAData, []byte(fixtures.TLSCACertPEM))
+			require.NoError(t, tt.args.validateBearerToken(got.getKubeRestConfig().BearerToken))
+			require.Equal(t, got.getTargetAddr(), tt.wantAddr)
+			require.NoError(t, got.close())
+		})
+	}
+}
+
+type eksMockClient struct {
+	eksiface.EKSAPI
+	cluster types.KubeCluster
+	t       *testing.T
+}
+
+func (e *eksMockClient) DescribeClusterWithContext(_ aws.Context, req *eks.DescribeClusterInput, _ ...request.Option) (*eks.DescribeClusterOutput, error) {
+	require.Equal(e.t, e.cluster.GetAWSConfig().Name, *req.Name)
+	return &eks.DescribeClusterOutput{
+		Cluster: &eks.Cluster{
+			Endpoint: aws.String("https://api.eks.us-west-2.amazonaws.com"),
+			Name:     req.Name,
+			CertificateAuthority: &eks.Certificate{
+				Data: aws.String(base64.RawStdEncoding.EncodeToString([]byte(fixtures.TLSCACertPEM))),
+			},
+		},
+	}, nil
+}
+
+type stsMockClient struct {
```

Comment:
We have lib/cloud/mocks/aws.go. Can we use STS mock we already have there, and move EKS mock there as well?